core/injector/application_injector.hpp

```diff
@@ -490,7 +490,7 @@ namespace kagome::injector {
         di::bind<network::Router>.template to<network::RouterLibp2p>(),
         di::bind<network::SyncProtocolClient>.template to<consensus::SynchronizerImpl>(),
         di::bind<network::SyncProtocolObserver>.template to<consensus::SynchronizerImpl>(),
-        di::bind<runtime::WasmMemory>.template to<runtime::WasmMemoryImpl>(),
+        di::bind<runtime::WasmMemory>.template to<runtime::binaryen::WasmMemoryImpl>(),
```

Comment:
actually this bind is redundant. There is no constructor accepting WasmMemory. Remove it from here and from linking libraries for injector